### PR TITLE
Add doc for "How To Run Against S3 Locally"

### PR DESCRIPTION
**Pull request by jim**
## Description

This PR adds a first draft of instructions on running the app against a live S3 instance for development purposes.

The commands documented aren't what we want them to be eventually, but there are some issues preventing the intended configuration from working properly. Once we get those ironed out, this PR should be updated accordingly.

### CODE REVIEW

**Comment by chrisgilmerproj** [comment]
Might be nice to reference https://github.com/transcom/ppp-infra/tree/master/transcom-ppp which is where we configure `app-devlocal` to make the S3 bucket.

Also, we need another one of these READMEs for using SES (something to think about).

File: docs/how-to/run-against-s3-locally.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+## Upload Files to S3
+
+The environment variable `STORAGE_BACKEND` specifies if files should be stored on `s3`, on the `local` filesystem, or in `memory`. The default value is `local` in development and `s3` when the app is running in any deployed environment.
+
+### Using the Devlocal Bucket
+
+Assuming your AWS credentials are setup properly, this command will configure the app to upload to the `transcom-ppp-app-devlocal-us-west-2` S3 bucket:
+
+```console
+$ env STORAGE_BACKEND=s3 aws-vault exec $AWS_PROFILE -- bin/webserver
```
Comment (chrisgilmerproj) [inline docs/how-to/run-against-s3-locally.md]: I think you can do `env STORAGE_BACKEND=s3 aws-vault exec $AWS_PROFILE -- make server_run_standalone`

**Comment by jacquelineIO** [comment]
This is a lot of what I had to put into my PR so that I know how to do and whoever is approving it, so I like 👍   https://github.com/transcom/mymove/pull/1620

File: docs/how-to/run-against-s3-locally.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+### Using the Devlocal Bucket
+
+Assuming your AWS credentials are setup properly, this command will configure the app to upload to the `transcom-ppp-app-devlocal-us-west-2` S3 bucket:
+
+```console
+$ env STORAGE_BACKEND=s3 make server_run_standalone
+```
+
+_Please note that this does not use our usual setup to automatically reload the server when a change is made._
```
Comment (chrisgilmerproj) [inline docs/how-to/run-against-s3-locally.md]: It does use `gin` but it doesn't use `entr` to watch the swagger files.  So it still ought to reload on *most* changes.

**Comment by jacquelineIO** [comment]
This looks good to me. Might consider adding how to remove the file from `s3` if this was some testing being done. For my PR, I asked for the reviewer to delete the file. I don't know if that's necessary or not, but worth adding if you think it makes sense.

File: docs/how-to/run-against-s3-locally.md (hunk 2 above)
Comment (jim) [reply docs/how-to/run-against-s3-locally.md]: Oh, nice. That's not bad at all, then.

**Pull request merged**